pkg/oc/cli/admin/ipfailover/ipfailover.go

```diff
@@ -60,189 +57,245 @@ var (
+	configList := []runtime.RawExtension{
+		{
+			Object: &corev1.ServiceAccount{
+				// this is ok because we know exactly how we want to be serialized
+				TypeMeta:   metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "ServiceAccount"},
+				ObjectMeta: metav1.ObjectMeta{Name: o.ConfigOptions.ServiceAccount},
+			},
+		},
 	}
 
 	list.Items = append(configList, list.Items...)
 
-	if options.Action.ShouldPrint() {
-		return print.VersionedPrintObject(kcmdutil.PrintObject, cmd, options.Action.Out)(list)
+	// TODO: stop treating --output formats as --dry-run
+	if o.PrintFlags.OutputFormat != nil && len(*o.PrintFlags.OutputFormat) > 0 && *o.PrintFlags.OutputFormat != "name" {
+		return o.Printer.PrintObj(list, o.Out)
 	}
 
-	if errs := options.Action.WithMessage(fmt.Sprintf("Creating IP failover %s", name), "created").Run(list, namespace); len(errs) > 0 {
-		return kcmdutil.ErrExit
+	created, errs := o.createResources(list, o.DryRun)
+
+	// print what we have created first, then return a potential set of errors
+	if err := o.Printer.PrintObj(created, o.Out); err != nil {
+		errs = append(errs, err)
 	}
-	return nil
+
+	return kerrors.NewAggregate(errs)
 }
 
-func validateServiceAccount(client securityclientinternal.Interface, ns string, serviceAccount string) error {
-	sccList, err := client.Security().SecurityContextConstraints().List(metav1.ListOptions{})
-	if err != nil {
-		if !errors.IsUnauthorized(err) {
-			return fmt.Errorf("could not retrieve list of security constraints to verify service account %q: %v", serviceAccount, err)
-		}
+func (o *IPFailoverConfigCmdOptions) createResources(list *corev1.List, dryRun bool) (*unstructured.UnstructuredList, []error) {
+	errors := []error{}
+	created := &unstructured.UnstructuredList{
+		Object: map[string]interface{}{
+			"kind":       "List",
+			"apiVersion": "v1",
+			"metadata":   map[string]interface{}{},
+		},
 	}
 
-	for _, scc := range sccList.Items {
-		if scc.AllowPrivilegedContainer {
-			for _, user := range scc.Users {
-				if strings.Contains(user, serviceAccount) {
-					return nil
-				}
-			}
+	for _, item := range list.Items {
+		var err error
+		unstructuredObj := &unstructured.Unstructured{}
+		unstructuredObj.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(item.Object)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		namespace := o.Namespace
+		if len(unstructuredObj.GetNamespace()) > 0 {
+			namespace = unstructuredObj.GetNamespace()
 		}
+
+		mapping, err := o.RESTMapper.RESTMapping(unstructuredObj.GroupVersionKind().GroupKind(), unstructuredObj.GroupVersionKind().Version)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if mapping.Scope.Name() == meta.RESTScopeNameRoot {
+			namespace = ""
+		}
+
+		if dryRun {
+			created.Items = append(created.Items, *unstructuredObj)
+			continue
+		}
+
+		_, err = o.DynamicClient.Resource(mapping.Resource).Namespace(namespace).Create(unstructuredObj)
```

Comment:
Please use short if, where possible, this is a perfect candidate. Btw. if you don't know that yet @pweil- calls me short-if :policeman:

Reply:
+1